src/Microsoft.Azure.ServiceBus/MessageBatchReceivePump.cs

```diff
@@ -0,0 +1,277 @@
+                    {
+                        this.maxConcurrentCallsSemaphoreSlim.Release();
+                        MessagingEventSource.Log.MessageReceiverPumpTaskStop(this.messageReceiver.ClientId, this.maxConcurrentCallsSemaphoreSlim.CurrentCount);
+                    }
+                }
+            }
+        }
+
+        async Task MessageBatchDispatchTaskInstrumented(IList<Message> messages)
+        {
+            IEnumerable<Activity> activities = this.diagnosticSource.ProcessStart(messages);
+            Task processTask = null;
+            try
+            {
+                processTask = MessageBatchDispatchTask(messages);
+                await processTask.ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                this.diagnosticSource.ReportException(e);
+                throw;
+            }
+            finally
+            {
+                this.diagnosticSource.ProcessStop(activities, processTask?.Status);
+            }
+        }
+
+        async Task MessageBatchDispatchTask(IList<Message> messages)
+        {
+            CancellationTokenSource renewLockCancellationTokenSource = null;
+            Timer autoRenewLockCancellationTimer = null;
+
+            MessagingEventSource.Log.MessageBatchReceiverPumpDispatchTaskStart(this.messageReceiver.ClientId, messages);
+
+            if (this.ShouldRenewLock())
+            {
+                renewLockCancellationTokenSource = new CancellationTokenSource();
+                TaskExtensionHelper.Schedule(() => this.RenewMessageLockTask(messages, renewLockCancellationTokenSource.Token));
+
+                // After a threshold time of renewal('AutoRenewTimeout'), create timer to cancel anymore renewals.
+                autoRenewLockCancellationTimer = new Timer(this.CancelAutoRenewLock, renewLockCancellationTokenSource, this.registerHandlerOptions.MaxAutoRenewDuration, TimeSpan.FromMilliseconds(-1));
+            }
+
+            try
+            {
+                MessagingEventSource.Log.MessageBatchReceiverPumpUserCallbackStart(this.messageReceiver.ClientId);
+                await this.onMessageBatchCallback(messages, this.pumpCancellationToken).ConfigureAwait(false);
+
+                MessagingEventSource.Log.MessageBatchReceiverPumpUserCallbackStop(this.messageReceiver.ClientId);
+            }
+            catch (Exception exception)
+            {
+                MessagingEventSource.Log.MessageBatchReceiverPumpUserCallbackException(this.messageReceiver.ClientId, exception);
+                await this.RaiseExceptionReceived(exception, ExceptionReceivedEventArgsAction.UserCallback).ConfigureAwait(false);
+
+                // Nothing much to do if UserCallback throws, Abandon message and Release semaphore.
+                if (!(exception is MessageLockLostException))
+                {
+                    await this.AbandonMessagesIfNeededAsync(messages).ConfigureAwait(false);
+                }
```

Comment:
You are abondoning every message because one message ran into lock lost exception?